src/renderers/testing/ReactTestRendererFiber.js

```diff
@@ -217,6 +220,29 @@ var TestRenderer = ReactFiberReconciler({
     setTimeout(fn, 0, {timeRemaining: Infinity});
   },
 
+  mockComponent(component: Fiber, rootContainer: Container) {
+    invariant(
+      component._unmockedType === null,
+      'Trying to mock an already mocked component',
+    );
+    const mockedFn = rootContainer.createComponentMock({
+      type: component.type,
+      props: component.pendingProps,
+    });
+    invariant(
+      typeof mockedFn === 'function',
+      'createComponentMock() must return a function. Found %s instead.',
+      typeof mockedFn,
+    );
+    if (mockedFn !== component.type) {
+      component._unmockedType = component.type;
+      component.type = mockedFn;
```

Comment:
We should never mutate the Fiber here. We should instead return the mock and let the algorithm initialize it in the appropriate place.